"""Создайте собственный класс-исключение,
обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем.
При вводе пользователем нуля в качестве делителя
программа должна корректно обработать эту ситуацию
и не завершиться с ошибкой."""

class MyError(Exception):

    def __init__(self, text=''):
        self.text = text


var_1 = int(input('Введите первое число: '))
var_2 = int(input('Введите второе число: '))

while True:
    try:
        if var_2 <= 0:
            raise MyError('Делить на 0 нельзя. Повторите ввод')
    except MyError as err:
        print(err)
        var_2 = int(input('Повторите ввод второго числа: '))
    else:
        break

print(var_1 / var_2)
